test/workshop_test.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+// TestWorkshops searches the parent directory for directories that
+// contain a README.md file. It then parses those for START/END tags
+// and runs the bash scripts between those tags. It only supports bash
+// actions at this time.
+func TestWorkshops(tt *testing.T) {
+	workshops, err := findWorkshops()
+	if err != nil {
+		tt.Log("Error finding workshop READMEs")
+		tt.Fatal(err.Error())
+	}
+
+	for _, ws := range workshops {
+		text, err := ioutil.ReadFile(ws)
+		if err != nil {
+			tt.Logf("Could not read workshop: %v", ws)
+			tt.Fatal(err.Error())
+		}
+
+		actions, err := GetActions(bytes.NewReader(text))
+		if err != nil {
+			tt.Logf("Could not parse actions in workshop: %v", ws)
+			tt.Fatal(err.Error())
+		}
+
+		tt.Logf("Running scripts in: %v", ws)
+		for _, action := range actions {
+			switch action.ActionType {
+			case "bash":
+				err := os.Chdir(filepath.Dir(ws))
+				if err != nil {
+					tt.Logf("Error changig to workshop dir: %v",
+						filepath.Dir(ws))
+					tt.Fatal(err.Error())
+				}
+				runBash(tt, action.Commands)
+			default:
+				tt.Errorf("Invalid action type found in workshop: %v",
+					action.ActionType)
+			}
+		}
+	}
+}
+
+// Run a series of bash commands
+func runBash(tt *testing.T, commands []string) {
+	for _, line := range commands {
+		cmdS := strings.Fields(line)
+		cmd := exec.Command(cmdS[0], cmdS[1:]...)
+		var sout bytes.Buffer
+		var serr bytes.Buffer
+		cmd.Stdout = &sout
+		cmd.Stderr = &serr
+		err := cmd.Run()
+		if err != nil {
+			tt.Errorf("Command failed: %v", line)
+			tt.Errorf("Error: %v", err)
+			tt.Errorf("Std Out: %v", sout.String())
+			tt.Errorf("Std Err: %v", serr.String())
```

Comment:
Don't want a fatal here, we want to break on this script and move to the next directory/workshop.

CombinedOutput sounds good, I think it is usually more helpful to see stderr interleaved with stdout.